website/docs/Installation.md

```diff
@@ -1,10 +1,50 @@
 # Installation
 
-## Setup Virtual Environment
+## Option 1: Install and Run AutoGen in Docker
 
-When not using a docker container, we recommend using a virtual environment to install AutoGen. This will ensure that the dependencies for AutoGen are isolated from the rest of your system.
+[Docker](https://www.docker.com/) is a containerization platform that simplifies the setup and execution of your code. A properly built docker image could provide isolated and consistent environment to run your code securely across platforms. One option of using AutoGen is to install and run it in a docker container. You can do that in [Github codespace](https://codespaces.new/microsoft/autogen?quickstart=1) or follow the instructions below to do so.
 
-### Option 1: venv
+#### Step 1. Install Docker.
+
+Install docker following [this instruction](https://docs.docker.com/get-docker/).
+
+For Mac users, alternatively you may choose to install [colima](https://smallsharpsoftwaretools.com/tutorials/use-colima-to-run-docker-containers-on-macos/) to run docker containers, if there is any issues with starting the docker daemon.
+
+#### Step 2. Build a docker image
+
+AutoGen provides [dockerfiles](https://github.com/microsoft/autogen/tree/main/samples/dockers/) that could be used to build docker images. Use the following command line to build a docker image named `autogen_img` (or other names you prefer) from one of the provided dockerfiles named `Dockerfile.base`:
+
+```
+docker build -f samples/dockers/Dockerfile.base -t autogen_img https://github.com/microsoft/autogen.git#main
+```
+which includes some common python libraries and essential dependencies of AutoGen, or build from `Dockerfile.full` which include additional dependencies for more advanced features of AutoGen with the following command line:
+
+```
+docker build -f samples/dockers/Dockerfile.full -t autogen_full_img https://github.com/microsoft/autogen.git
+```
+Once you build the docker image, you can use `docker images` to check whether it has been created successfully.
+
+#### Step 3.  Run applications built with AutoGen from a docker image.
+
+**Mount your code to the docker image and run your application from there:** Now suppose you have your application built with AutoGen in a main script named `twoagent.py` ([example](https://github.com/microsoft/autogen/blob/main/test/twoagent.py)) in a folder named `myapp`. With the command line below, you can mont your folder and run the application in docker.
+
+```python
+# Mount the local folder `myapp` into docker image and run the script named "twoagent.py" in the docker.
+docker run -it -v `pwd`/myapp:/myapp autogen_img:latest python /myapp/main_twoagent.py
```

Comment:
I might be wrong because I didn't test it, but I believe that `git` will not work inside the container the way it has been set up. If you use ssh for authentication, then commands such as `git pull` and `git push` will look into `~/.ssh/` for auth data. We could provide instructions for mapping `.ssh` folder in host to container to avoid this.